<a href="https://colab.research.google.com/github/JuliaBalakina/Python_ALATA_2023/blob/main/2023_ALaTA_02_bool_if.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Variables

A **variable** is a name that refers to a value.

An **assignment statement** creates new variables and gives them values:

x = 2

message = 'And now for something completely different'

This example makes two assignments. The first assigns the integer 2 to x; the second assigns a string to a new variable
named "message".

To display the value of a variable, you can use a print statement:
>print(x)

>2

>print(message)

>'And now for something completely different'

The type of a variable is the type of the value it refers to.

>type(message)

><class 'str'>

>type(x)

><class 'int'>

Variable names can be arbitrarily long. They can contain both letters and numbers,
but **they cannot start with a number**. It is legal to use uppercase letters, but it is
a good idea to begin variable names with a lowercase letter.
The underscore character ( _ ) can appear in a name. It is often used in names with
multiple words, such as my_name or airspeed_of_unladen_swallow.

*If you give a variable an illegal name, you get a syntax error:*

>76trombones = 'big parade'

>SyntaxError: invalid syntax

Python reserves **35 keywords which cannot be used as variable
names:**

> and  del  from  None  True

>as elif global nonlocal try

>assert else if not while

>break except import or with

>class False in pass yield

>continue finally is raise async

>def for lambda return await



In [ ]:
a = 2
b = 3
a = b ## a=3
b = a ## b=3
print(a,b)

3 3


In [ ]:
a=2
b=3
b=a
print(b)
print(a)

2
2


### Changing value

>gamma = 1

>gamma = 2

Suppose we have these two lines of Python.

Python let you set a name for a value and then change it.




In [ ]:
a = 2
b = 3
temp = b ## 3
b = a ##2
a = temp ## 3
print(a,b)


3 2


In [ ]:
student1 = 'Masha'
student2 = 'Dasha'
temp = "Masha"
student1 = student2
student2 =temp
print(student1,student2)

Dasha Masha


In [ ]:
name = 'John'
surname = 'Black'
print (name, surname)

John Black


In [ ]:
name = 'John'
surname = 'Black'
tempor = name #John
name = surname # Black
surname = tempor
print (name, surname)


Black John


# Task
3 people participated in a song contest: Jack,Bill and Max. The Jury made a mistake when assigning the places:

place_1 = 'Jack'
place_2 = 'Bill'
place_3 = 'Max'

 It should go like this:
place_1 = 'Bill'
place_2 = 'Max'
place_3 = 'Jack'

Write a code which will change the order.
Your output:

print(place_1,place_2,place_3)

Bill,Max,Jack

In [1]:
place_1 = 'Jack'
place_2 = 'Bill'
place_3 = 'Max'



### Syntactic sugar

>x += 10  (x = x + 10)

>x -= 10                     (x = x - 10)

>x *= 10                     (x = x * 10)

>x /= 10      (x = x / 10)

>x * *= 10                   (x = x * * 10)

>x %= 10                    (x = x % 10)



In [ ]:
x = 3
x*=5 ## x = x*5
print(x)

15

In [ ]:
x=2
x**=2
print(x)

4


### Deleting a name

To delete the attachment we use the Python command “del”.

x = 3

del x


In [ ]:
x = 3
del x
print (x)

NameError: name 'x' is not defined

### Boolean Expressions


**True** and **False** are (the only) two values of a special Python type called a “Boolean” used for
recording whether something is true or not.

You can turn something into a boolean variable using the **bool ()** function.

If you convert numbers then - 0 will turn into False, everything else - True.

In [ ]:
print(type(True), type(False))

<class 'bool'> <class 'bool'>


In [ ]:
print(bool(0))
print(bool(23))
print(bool(-10))

False
True
True


An empty string is converted to False, all other strings to True.

The boolean variable can be converted to **int**  - zero and one.

In [ ]:
print(bool())
print(bool('k'))

False
True


In [ ]:
print(int(True))
print(int(False))

1
0


### Comparison Operators

Boolean expressions ask a question and produce a **Yes** or **No** result which we use to control program flow

Boolean expressions using comparison operators evaluate to True / False or Yes / No

Comparison operators look at variables but do not change the variables


    == (Equal to)
    != (Not equal)
    > (Greater than)
    < (Less than)
    >= (Greater than or equal to)
    <= (Less than or equal to)



In [ ]:
print(1 == 1)
print(1 != '1')
c = 1 > 3
print(c)
print(type(c))
x = 5
print(1 < x <= 5) # 1<x and x<=5

True
True
False
<class 'bool'>
True


### Logical operators

      AND - expression is true only when both sides are true, otherwise it is false
      OR (or) - expression is false only when both parts are false, otherwise it is true
      negation (not) - This takes only one input and “flips” it. True becomes False and vice versa.



In [ ]:
print((1 == 1) and ('1' == 1))
print(('1' == 1) or (1 == 1))
print(not(1 == 1))
print(((1 == 1) or ('1' == 1)) and (2 == 2))

False
True
False
True


# Task

Prompt the user for an integer.

Create an expression checking whether the user's input is larger than 0 and smaller than 100.

Output should return "True" or "False"

# Task

Prompt the user for three integers.

Create an expression comparing them.

If two of them are equal - "True"

If they are all equal - "True"


### Task

Vasya went to Italy to study. The only store in the city is open **from 6 am to 8 am and from 4 pm to 5 pm**. Vasya has not been able to get to the store for several days and is suffering from hunger. He can come to the store at X o'clock. If the store is open at X hours, then output should be ***True***, and if it is closed, output -  ***False***.

Input contains an integer X ( from 0 to 23 )

Output format: True or False

Example:

Input data: 16

Output of the program: True

### Conditional execution

In order to write useful programs, we almost always need the ability to check conditions
and change the behavior of the program accordingly. Conditional statements
give us this ability. The simplest form is the if statement:

>if x > 0 :

>>print('x is positive')

The boolean expression after the if statement is called ***the condition***. We end the
if statement with a colon character (:) and the line(s) after the if statement are
indented.  
If the logical condition is true, then the indented statement gets executed. If the
logical condition is false, the indented statement is skipped.

In [11]:
x=int(input())
if x>0:
    print(x)

1
1


In [ ]:
x = 50
if x == 5 :
    print('Equals 5')
if x > 4 :
    print('Greater than 4')
if  x >= 5 :
    print('Greater than or Equals 5')
if x < 6 :
    print('Less than 6')
if x <= 5 :
    print('Less than or Equals 5')
if x != 6 :
    print('Not equal 6')

Greater than 4
Greater than or Equals 5
Not equal 6


### Alternative execution  
A second form of the if statement is alternative execution, in which there are two  
possibilities and the condition determines which one gets executed. The syntax
looks like this:  
>if x%2 == 0 :
>>print('x is even')


>else :                  
>>print('x is odd')

In [ ]:
x = int(input())
if x%2 == 0 :
    print('x is even')
else :
    print('x is odd')

3
x is odd


### Chained conditionals  

Sometimes there are more than two possibilities and we need more than two
branches. One way to express a computation like that is a chained conditional:


>if x < y:
>>print('x is less than y')

>elif x > y:
>>print('x is greater than y')

>else:
>>print('x and y are equal')

elif is an abbreviation of “else if.” Again, exactly one branch will be executed.
There is no limit on the number of elif statements. If there is an else clause, it
has to be at the end, but there doesn’t have to be one.  
Each condition is checked in order. If the first is false, the next is checked, and so
on. If one of them is true, the corresponding branch executes, and the statement
ends. Even if more than one condition is true, only the first true branch executes.

In [ ]:
x = int(input())
y = int (input())
if x <= y:
    print('x is less than y')
elif x > y:
    print('x is greater than y')
else:
    print('x and y are equal')



3
3
x is less than y


### Task
Masha and Petya write a number down on a piece of paper, which they give to Kolya. Kolya announces the name of the winner ( the person with the largest number wins) or announces a draw if the numbers are equal.

INPUT FORMAT:

     A - an integer from 0 to 1000 (the number that Masha has in mind)
     B - an integer from 0 to 1000 (the number that Petya meant)
     Each is entered on a new line.

OUTPUT FORMAT:

     String "Masha", "Petya" or "Draw", the answer to the problem

### Nested conditionals  
One conditional statement can also be nested within another. We could have written the
three-branch example like this:

if x == y:    
>print('x and y are equal')

else:  

>if x < y:

>>print('x is less than y')

>else:

>>print('x is greater than y')

In [12]:
x = int(input()) ##The outer conditional contains two branches. The first branch contains a simple statement. The second branch contains another if statement, which has two branches of its own.
y = int (input())
if x == y:
    print('x and y are equal')
else:
    if x < y:
        print('x is less than y')
    else:
        print('x is greater than y')



4
5
x is less than y


Logical operators often provide a way to simplify nested conditional statements.
For example, we can rewrite the following code using a single conditional:  

 if 0 < x and x < 10:
    
    print('x is a positive single-digit number.')


In [ ]:
x = int(input())
if 0 < x < 10:
    print('x is a positive single-digit number.')
else:
    pass

0


### Task

You are given a number and have to make it negative. But maybe the number is already negative?

Example:

input(1);  # output (-1)


input(-5); # output (-5)


input(0);  # output (0)



### Task  (Home assignment)
Write a program to prompt for a score between 0.0 and
1.0. If the score is out of range, print ' Bad score'. If the score is
between 0.0 and 1.0, print a grade using the following table:

0.9<=x<=1.0 - A

0.8<=x<0.9 - B

0.7<=x<0.8 - C

0.6<=x<0.7 - D

x< 0.6 - F

Example:

Enter score: 0.95  

A  



### Task

There is a special offer in the store:

     For all products cheaper than 1000 rubles 15% discount
     For all products more expensive than 1000, but cheaper than 5000 rubles, a 20% discount
     For all products more expensive than 5000 rubles a 25% discount

Input  

Non-negative integer - product price in rubles

Output  

Non-negative integer - discount for goods in rubles

### Task

Create a programm that gives a personalized greeting.

Input: name and the name of the owner ( 2 names)

Use conditionals to print the proper message:
    

name equals the name of the owner  ---	'Hello boss'

otherwise ---	'Hello guest'

# Task

Write a programme that prompts the user for  two arguments (salary, bonus). Salary will be an integer, and bonus "Yes" or "No".

If bonus is Yes, the salary should be multiplied by 10.

If bonus is No, the fatcat did not make enough money and must receive only his stated salary.

# Task

Create a program that checks if a number n is divisible by two numbers x AND y.

All inputs are strictly positive numbers.

Examples:

1) n =   3, x = 1, y = 3 =>  true because   3 is divisible by 1 and 3

2) n =  12, x = 2, y = 6 =>  true because  12 is divisible by 2 and 6

3) n = 100, x = 5, y = 3 => false because 100 is not divisible by 3

4) n =  12, x = 7, y = 5 => false because  12 is neither divisible by 7 nor 5

# Extra exercises with solutions

https://www.practicepython.org/exercises/

https://holypython.com/beginner-python-exercises/

https://pythonbasics.org/exercises/

https://pynative.com/python-basic-exercise-for-beginners/

https://www.w3resource.com/python-exercises/python-basic-exercises.php

